Story 1:

SamuelConnor opened the pull request:
CRUD endpoints
mockMVC tests for all endpoints.

Kieransullivan-imanage commented on src/main/java/com/example/demo/controllers/LocationController.java:
```diff
@@ -0,0 +1,59 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/location")
```
just use /api here

Kieransullivan-imanage commented on src/main/java/com/example/demo/controllers/LocationController.java:
```diff
@@ -0,0 +1,59 @@
+    @Autowired
+    LocationService locationService;
+
+    @PostMapping(path = "/createLocation")
```
The path would just be the thing you're creating, so just /location

Kieransullivan-imanage commented on src/main/java/com/example/demo/controllers/LocationController.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @ResponseBody
+    @GetMapping(path = "/findLocation")
```
Because you're using HTTP GET, that implies 'find', 'fetch' type verbs. So you'd also just use /location here

Kieransullivan-imanage commented on src/main/java/com/example/demo/controllers/LocationController.java:
```diff
@@ -0,0 +1,59 @@
+
+    @ResponseBody
+    @GetMapping(path = "/findLocation")
+    public LocationEntity findById(@RequestParam("id") Long id)
```
The path will also have {id} in it because this get is filtered by id

Kieransullivan-imanage commented on src/main/java/com/example/demo/controllers/LocationController.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @ResponseBody
+    @GetMapping(path = "/allLocations")
```
This would be a GET mapping with a path of 'location', but without specifying a particular id to find

Kieransullivan-imanage commented on src/main/java/com/example/demo/controllers/LocationController.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @ResponseBody
+    @GetMapping(path = "/allLocationsOrdered")
```
The URL query parameter can specify the sort and so you should not need a specific method for ordering - ordering can apply to all the GET URLs

Kieransullivan-imanage commented on src/main/java/com/example/demo/controllers/LocationController.java:
```diff
@@ -0,0 +1,59 @@
+        return locationService.findAllLocationsAscending();
+    }
+
+    @PutMapping(path = "/updateLocation/{id}")
```
again, just /location/{id}

Kieransullivan-imanage reviewed (changes requested):
Still reviewing...but some changes required so far.

Kieransullivan-imanage commented on src/main/java/com/example/demo/controllers/LocationController.java:
```diff
@@ -0,0 +1,55 @@
+
+    @ResponseBody
+    @GetMapping(path = "/location")
+    public List<LocationEntity> allLocationsOrdered(@RequestParam (value = "page", defaultValue = "0") int page,
```
Does the Pageable page parameter here not get you all of this?

SamuelConnor replied:
I couldn't find a default constructor for Pageable:
"Cannot construct instance of `org.springframework.data.domain.Pageable` (no Creators, like default construct, exist)"

Kieransullivan-imanage replied:
I am sure there is a better way. I'll do some digging :-)

Kieransullivan-imanage commented on src/main/java/com/example/demo/controllers/LocationController.java:
```diff
@@ -0,0 +1,55 @@
+    @GetMapping(path = "/location/{id}")
+    public LocationEntity findById(@PathVariable Long id)
+    {
+    return locationService.findLocationById(id);
```
Indentation

Kieransullivan-imanage commented on src/main/java/com/example/demo/services/LocationService.java:
```diff
@@ -0,0 +1,57 @@
+
+    public List<LocationEntity> findAllLocations()
+    {
+        return (List<LocationEntity>) locationEntityRepository.findAll();
```
findAllLocations and findAllLocationsOrdered can be replaced with just the findAllLocationsOrdered method as findAllLocations is just the ordered version with an unpaged and unsorted parameter?

SamuelConnor replied:
Removed, in next story, missed it this one.

Kieransullivan-imanage commented on src/test/groovy/com/example/demo/LocationTests.groovy:
```diff
@@ -0,0 +1,179 @@
+            def json = new JsonSlurper().parseText(result)
+        then: "Count the number of objects"
+            json.size() == locationEntityRepository.count()
+            json.size() == 3
```
Better to check the three ids.

Kieransullivan-imanage reviewed (changes requested):
Looks good. Just some minor comments

SamuelConnor committed:
Story 1:
-Fixed further issues from PR
-Improved unit test for finding All Locations

Kieransullivan-imanage commented on src/main/java/com/example/demo/services/LocationService.java:
```diff
@@ -20,22 +20,24 @@ public LocationEntity createLocation(LocationEntity location)
+    public List<LocationEntity> findAllLocationsOrdered(int page, int size, String criteria, boolean ascending)
     {
-        Pageable pageable = PageRequest.of(0, 20, Sort.by("country").ascending());
+        Pageable pageable = PageRequest.of(page, size, Sort.by(criteria).ascending());
```
Do you need to validate 'criteria'? Could this introduce a security issue?

Kieransullivan-imanage reviewed (changes requested):
Validation of criteria is the only real concern

SamuelConnor committed:
Story 1:
-Changed find locations method to take in parameter of pageable
-Changed unit test to suit new parameter

Kieransullivan-imanage reviewed (approved):
Looks better

SamuelConnor committed:
Merge branch 'master' into samc-spring-1
